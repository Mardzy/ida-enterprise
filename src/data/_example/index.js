import reducer from "./reducers";

export { default as duckEpics } from "./epics";
export { default as duckTypes } from "./types";

export default reducer;