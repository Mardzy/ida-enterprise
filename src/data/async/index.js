import { async as reducer } from "./reducers";

export default reducer;