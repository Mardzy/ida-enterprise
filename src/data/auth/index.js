import authReducer from "./reducers";

export { default as authEpics } from "./epics";
export { default as authTypes } from "./types";
export { default as authActions } from "./actions";

export default authReducer;