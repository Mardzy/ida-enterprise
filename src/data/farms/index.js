import farmsReducer from "./reducers";

export { default as farmEpics } from "./epics";
export { default as farmTypes } from "./types";

export default farmsReducer;