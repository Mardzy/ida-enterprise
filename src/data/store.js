import { fromJS } from "immutable";
import { createEpicMiddleware } from "redux-observable";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import persistState from "redux-localstorage";
import { routerMiddleware } from "connected-react-router/immutable";

import { INITIAL_STATE } from "./initial-state";
import { rootReducer, history } from "./reducer";
import { rootEpic } from "./epic";
// import { gaMiddleware } from "data/config/analytics";
import { serialize, deserialize, merge, slice } from "./config/persistent-store";

/**
 * Construct an epic middleware.
 * @type {EpicMiddleware<Action, any, any, Action>}
 */
const epicMiddleware = createEpicMiddleware();

/**
 * Middleware to apply in the store.
 * @type {*[]}
 */
const middleware = [
  // gaMiddleware,
  epicMiddleware,
  routerMiddleware(history)
];

/**
 * Configuration for keeping all the redux state in localstorage.
 * @type {{key: string, slicer: slice, serialize: serialize, deserialize: deserialize, merge: merge}}
 */
const persistStateConfig = {
  key: "partner-dashboard",
  slicer: slice,
  serialize, deserialize, merge
};

/**
 * Create a Redux store.
 * @param {Object} initialState
 * @return {Store<any>}
 */
const configureStore = (initialState = {}) => createStore(
  rootReducer,
  INITIAL_STATE.merge(fromJS(initialState)),
  composeWithDevTools({})(
    applyMiddleware(...middleware),
    persistState(["auth", "farms"], persistStateConfig)
  )
);

/**
 * Create a store with the reducers.
 */
const store = configureStore();

epicMiddleware.run(rootEpic);

export default store;
export { history, epicMiddleware, configureStore };