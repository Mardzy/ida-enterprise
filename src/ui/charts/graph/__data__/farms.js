import { fromJS } from "immutable";

export default fromJS([{
  "id": 2,
  "farmName": "Infinita Dairies",
  "geoCoords": "31.913145,73.047907"
}, {
  "id": 3,
  "farmName": "Ormsby Field",
  "geoCoords": "52.536720,4.659249"
}, {
  "id": 4,
  "farmName": "Koningshoeve",
  "geoCoords": "52.405053,4.698681"
}, {
  "id": 6,
  "farmName": "Maatschappij van Tunen",
  "geoCoords": "52.499715,4.705119"
}, {
  "id": 8,
  "farmName": "ForFarmers Farm",
  "geoCoords": "51.660578,5.947523"
}, {
  "id": 10,
  "farmName": "Dairy Campus",
  "geoCoords": "53.180656,5.756555"
}, {
  "id": 11,
  "farmName": "Gerrit van Tunen",
  "geoCoords": "52.538170,4.725746"
}, {
  "id": 12,
  "farmName": "Gebroeders Kea",
  "geoCoords": "52.215158,4.670464"
}, {
  "id": 13,
  "farmName": "MTS Kramer",
  "geoCoords": "52.474063,4.742218"
}, {
  "id": 15,
  "farmName": "Verstijnen Melkveebedrijf",
  "geoCoords": "51.338110,5.324489"
}, {
  "id": 17,
  "farmName": "Geko",
  "geoCoords": "52.466814,4.701745"
}, {
  "id": 19,
  "farmName": "Granja Guzman Vicente",
  "geoCoords": "39.784030, -0.180990"
}, {
  "id": 20,
  "farmName": "Stef Penders",
  "geoCoords": "51.095122, Â 5.826678"
}, {
  "id": 21,
  "farmName": "Winter Wonder Farm",
  "geoCoords": "52.3665158,4.8649626"
}, {
  "id": 22,
  "farmName": "Geurt Kets",
  "geoCoords": "52.3961138,5.4997788"
}, {
  "id": 23,
  "farmName": "Aitken",
  "geoCoords": "45.0838495,-71.8716957"
}, {
  "id": 24,
  "farmName": "Bles Dairy Farm",
  "geoCoords": "52.992854, 5.748187"
}, {
  "id": 25,
  "farmName": "Rutten",
  "geoCoords": "53.132302, 4.887862"
}, {
  "id": 26,
  "farmName": "Grupo Lala",
  "geoCoords": "25.693966, -103.292292"
}, {
  "id": 27,
  "farmName": "Smolders",
  "geoCoords": "51.449818, 5.273824"
}, {
  "id": 28,
  "farmName": "Uijttewaal",
  "geoCoords": "51.855475, 5.635604"
}, {
  "id": 29,
  "farmName": "van Houts",
  "geoCoords": "51.290765, 5.460927"
}, {
  "id": 30,
  "farmName": "Ricardo Buil",
  "geoCoords": "42.383955,0.180742"
}, {
  "id": 31,
  "farmName": "Romme",
  "geoCoords": "51.451814, 5.198752"
}, {
  "id": 32,
  "farmName": "Marcel Derks",
  "geoCoords": "51.645910, 5.874887"
}, {
  "id": 33,
  "farmName": "Crossroads",
  "geoCoords": "39.450773, -112.628368"
}, {
  "id": 34,
  "farmName": "Eikelenboom",
  "geoCoords": "51.562832, 5.756757"
}, {
  "id": 35,
  "farmName": "van Rossem",
  "geoCoords": "51.127714, 3.964970"
}, {
  "id": 36,
  "farmName": "Neue Hope",
  "geoCoords": "40.732183, -85.124303"
}, {
  "id": 37,
  "farmName": "Hart",
  "geoCoords": "33.126517, -81.881765"
}, {
  "id": 38,
  "farmName": "McCarty",
  "geoCoords": "39.764617, -101.463305"
}, {
  "id": 39,
  "farmName": "van der Linden",
  "geoCoords": "51.599274, 5.471139"
}, {
  "id": 40,
  "farmName": "Schotanus",
  "geoCoords": "52.427681, 5.010169"
}, {
  "id": 41,
  "farmName": "VOF Laarman",
  "geoCoords": "52.464362, 6.296307"
}, {
  "id": 42,
  "farmName": "Tankink",
  "geoCoords": "51.983294, 6.503081"
}, {
  "id": 43,
  "farmName": "van Gijssel",
  "geoCoords": "52.981527, 6.281763"
}]);