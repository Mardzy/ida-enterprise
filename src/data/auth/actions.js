import types from "./types";
import { createAsyncActions } from "../async/utils";

/**
 * Login actions created from types.LOGIN
 * @type {{start: Action, fulfilled: Action, rejected: Action}}
 */
const login = createAsyncActions(
  types.LOGIN,
  (username, password) => ({
    username,
    password
  }),
  response => response,
  response => response
);


/**
 * Logout action
 * @type {Action}
 * @memberOf module:Actions
 * @return {{type: string}}
 */
const logout = () => ({
  type: types.LOGOUT.PENDING
});

/**
 * Actions created from types.REQUEST_USER
 * @type {{start: Action, fulfilled: Action, rejected: Action}}
 */
const requestUser = createAsyncActions(
  types.REQUEST_USER,
  username => ({
    email: username
  }),
  response => response,
  response => response
);

export default {
  login,
  logout,
  requestUser
};

