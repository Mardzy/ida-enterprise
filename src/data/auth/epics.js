import _ from "lodash";
import { of } from "rxjs";
import { ajax } from "rxjs/ajax";
import { mergeMap, mapTo, map, catchError } from "rxjs/operators";
import { ofType } from "redux-observable";

import { IDA_ENDPOINTS } from "../config/services";
import types from "./types";
import actions from "./actions";
import { Request } from "../api/ajax";

/**
 * Handle errors during login.
 * @param error
 * @return {Observable}
 */
const handleLoginError = error => of(actions.login.rejected(error));

/**
 * Register the token for further data requests, then pass on to the successful login handler.
 * @param request
 * @return {{type: string, payload: {response: Object}}}
 */
const handleLoginSuccess = request => {
  Request.registerToken(request.response.access_token);

  return actions.login.fulfilled(request.response);
};

/**
 * Login request config
 * @type {{method: string, url: string, crossDomain: boolean, headers: {}}}
 */
const loginRequestConfig = {
  method: "POST",
  url: `${IDA_ENDPOINTS.auth}/token`,
  crossDomain: true,
  headers: {
    "content-type": "application/x-www-form-urlencoded"
  }
};

/**
 * Log the user in, then pass the appropriate answer back to Redux.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 */
const login = action$ => action$.pipe(
  ofType(types.LOGIN.PENDING),
  mergeMap(action =>
    ajax(
      _.assign({}, loginRequestConfig, {
        body: _.assign({ grant_type: "password" }, action.payload)
      })
    ).pipe(
      map(handleLoginSuccess),
      catchError(handleLoginError)
    )
  )
);

/**
 * Logs the user out.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 * @return {Observable<{type: string}>}
 */
const logout = action$ => action$.pipe(
  ofType(types.LOGOUT.PENDING),
  mapTo({
    type: types.LOGOUT.FULFILLED
  })
);

/**
 * Handle errors during requesting user information.
 * @param error
 * @return {Observable}
 */
const handleUserRequestError = error => of(actions.requestUser.rejected(error));

/**
 * Store user information after successful request.
 * @param request
 * @return {{type: string, payload: {response: Object}}}
 */
const handleUserRequestSuccess = request => actions.requestUser.fulfilled(request.response);

/**
 * Request a user.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 * @return {Observable<AjaxResponse>}
 */
const requestUser = action$ => action$.pipe(
  ofType(types.REQUEST_USER.PENDING),
  mergeMap(action =>
    Request.GET(
      `${IDA_ENDPOINTS.auth}/Users/Find/EmailSimple/${encodeURIComponent(action.payload.email)}`
    ).pipe(
      map(handleUserRequestSuccess),
      catchError(handleUserRequestError)
    )
  )
);

export default {
  login,
  logout,
  requestUser
};