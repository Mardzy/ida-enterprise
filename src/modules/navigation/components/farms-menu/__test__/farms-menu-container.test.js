import { Provider } from "react-redux";
import { Menu } from "semantic-ui-react";

import store from "../../../../../data/store";
import FarmsMenuContainer from "../farms-menu-container";
import { authTypes } from "../../../../../data/auth";
import { farmTypes } from "../../../../../data/farms";

describe("The farms menu container", () => {

  it("renders nothing while loading", () => {
    const wrapper = mount(<Provider store={store}>
      <FarmsMenuContainer/>
    </Provider>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render a list of farms", () => {
    store.dispatch({
      type: authTypes.REQUEST_USER.FULFILLED,
      payload: {}
    });

    store.dispatch({
      type: farmTypes.FETCH_FARM_LIST.FULFILLED,
      payload: {
        value: [{
          FarmID: 1,
          farmName: "Lorem Ipsum"
        }, {
          FarmID: 2,
          farmName: "Diam Dolor"
        }]
      }
    });

    /**
     * @param farm
     * @return {*}
     */
    const renderItem = farm => <Menu.Item key={farm.get("FarmID")}>{farm.get("farmName")}</Menu.Item>;

    const wrapper = mount(<Provider store={store}>
      <FarmsMenuContainer renderItem={renderItem}/>
    </Provider>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});