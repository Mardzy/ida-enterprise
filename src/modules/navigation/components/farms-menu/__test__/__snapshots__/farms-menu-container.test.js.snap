// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The farms menu container render a list of farms 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(lifecycle(branch(Sink)))
    renderItem={[Function]}
  >
    <lifecycle(branch(Sink))
      fetch={[Function]}
      isLoading={false}
      items={
        Immutable.List [
          Immutable.Map {
            "FarmID": 1,
            "farmName": "Lorem Ipsum",
          },
          Immutable.Map {
            "FarmID": 2,
            "farmName": "Diam Dolor",
          },
        ]
      }
      renderItem={[Function]}
    >
      <branch(Sink)
        fetch={[Function]}
        isLoading={false}
        items={
          Immutable.List [
            Immutable.Map {
              "FarmID": 1,
              "farmName": "Lorem Ipsum",
            },
            Immutable.Map {
              "FarmID": 2,
              "farmName": "Diam Dolor",
            },
          ]
        }
        renderItem={[Function]}
      >
        <renderComponent(SubmenuWithLoader)
          fetch={[Function]}
          isLoading={false}
          items={
            Immutable.List [
              Immutable.Map {
                "FarmID": 1,
                "farmName": "Lorem Ipsum",
              },
              Immutable.Map {
                "FarmID": 2,
                "farmName": "Diam Dolor",
              },
            ]
          }
          renderItem={[Function]}
        >
          <SubmenuWithLoader
            fetch={[Function]}
            isLoading={false}
            items={
              Immutable.List [
                Immutable.Map {
                  "FarmID": 1,
                  "farmName": "Lorem Ipsum",
                },
                Immutable.Map {
                  "FarmID": 2,
                  "farmName": "Diam Dolor",
                },
              ]
            }
            renderItem={[Function]}
          >
            <MenuItem>
              <div
                className="item"
                onClick={[Function]}
              >
                <MenuMenu>
                  <div
                    className="menu"
                  >
                    <MenuItem
                      key="1"
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                      >
                        Lorem Ipsum
                      </div>
                    </MenuItem>
                    <MenuItem
                      key="2"
                    >
                      <div
                        className="item"
                        onClick={[Function]}
                      >
                        Diam Dolor
                      </div>
                    </MenuItem>
                  </div>
                </MenuMenu>
              </div>
            </MenuItem>
          </SubmenuWithLoader>
        </renderComponent(SubmenuWithLoader)>
      </branch(Sink)>
    </lifecycle(branch(Sink))>
  </Connect(lifecycle(branch(Sink)))>
</Provider>
`;

exports[`The farms menu container renders nothing while loading 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(lifecycle(branch(Sink)))>
    <lifecycle(branch(Sink))
      fetch={[Function]}
      isLoading={true}
      items={Immutable.List []}
    >
      <branch(Sink)
        fetch={[Function]}
        isLoading={true}
        items={Immutable.List []}
      >
        <Nothing
          fetch={[Function]}
          isLoading={true}
          items={Immutable.List []}
        />
      </branch(Sink)>
    </lifecycle(branch(Sink))>
  </Connect(lifecycle(branch(Sink)))>
</Provider>
`;
